silx/gui/plot/stats/stats.py

```diff
@@ -426,23 +439,41 @@ def __init__(self):
     def calculate(self, context):
         if context.kind in ('curve', 'histogram'):
             xData, yData = context.data
-            com = numpy.sum(xData * yData).astype(numpy.float32) / numpy.sum(
-                yData).astype(numpy.float32)
-            return com
+            deno = numpy.sum(yData).astype(numpy.float32)
+            if deno == 0.:
+                return 0.
+            else:
+                return numpy.sum(xData * yData).astype(numpy.float32) / deno
         elif context.kind == 'scatter':
-            xData = context.data[0]
-            values = context.values
-            com = numpy.sum(xData * values).astype(numpy.float32) / numpy.sum(
-                values).astype(numpy.float32)
-            return com
+            xData, yData, values = context.data
+            deno = numpy.sum(values).astype(numpy.float32)
+            if deno == 0.:
+                return 0., 0.
+            else:
+                xcom = numpy.sum(xData * values).astype(numpy.float32) / deno
+                ycom = numpy.sum(yData * values).astype(numpy.float32) / deno
+                return (xcom, ycom)
         elif context.kind == 'image':
             yData = numpy.sum(context.data, axis=1)
             xData = numpy.sum(context.data, axis=0)
             dataXRange = range(context.data.shape[1])
             dataYRange = range(context.data.shape[0])
+            xScale, yScale = context.scale
+            xOrigin, yOrigin = context.origin
 
-            ycom = numpy.sum(yData * dataYRange) / numpy.sum(yData)
-            xcom = numpy.sum(xData * dataXRange) / numpy.sum(xData)
+            denoY = numpy.sum(yData)
+            if denoY == 0.:
+                ycom = 0.
+            else:
+                ycom = numpy.sum(yData * dataYRange) / denoY
+                ycom = (ycom - yOrigin) * yScale + yOrigin
```

Comment:
I think it is more: ycom = ycom * yScale + yOrigin

Reply:
No, if i am correct we have to move to the data reference to apply the scale then to go back to the plot reference.